consensus/scheme/rolldpos/rolldposctx.go

```diff
@@ -578,11 +582,15 @@ func (ctx *rollDPoSCtx) Active() bool {
 	return ctx.active
 }
 
-///////////////////////////////////////////
+// /////////////////////////////////////////
 // private functions
-///////////////////////////////////////////
+// /////////////////////////////////////////
 
 func (ctx *rollDPoSCtx) mintNewBlock() (*EndorsedConsensusMessage, error) {
+	if ctx.round.Height() != ctx.lastMinted+1 {
```

Comment:
I would suggest we do the checking in https://github.com/iotexproject/iotex-core/pull/2436/files#diff-35b2b97589895506bc967676ee6f4ca9L333, and the parameter should be stored in a local db, like eManagerDB

Reply:
moreover, the condition could be a boolean `minted`, because it is per round.